sky/backends/backend_utils.py

```diff
@@ -2681,3 +2689,101 @@ def check_stale_runtime_on_remote(returncode: int, stderr: str,
                     f'not interrupted): {colorama.Style.BRIGHT}sky start -f -y '
                     f'{cluster_name}{colorama.Style.RESET_ALL}'
                     f'\n--- Details ---\n{stderr.strip()}\n')
+
+
+def get_endpoints(cluster: str,
```

Comment:
Good point - I've added `def endpoints()` to `sky.core`. However, I'm letting `serve/core.py` directly call `backend_utils.get_endpoints` because `from sky import core` in `serve/core.py` causes a circular import.